src/hooks/useApi.ts

```diff
@@ -0,0 +1,40 @@
+import { useEffect, useState } from 'react';
+
+const BASE_URL = '/api';
+
+/**
+ * Fetches resources from an API endpoint and returns them.
+ *
+ * @param path the path to the endpoint to fetch
+ * @returns the error and the result returned from the endpoint.
+ *    Only one of these will be set at any time.
+ */
+const useApi = <T>(path: string): [number | null, T | null] => {
+  const [result, setResult] = useState<T | null>(null);
+  const [error, setError] = useState<number | null>(null);
```

Comment:
Das hier sind die `useState()` Hooks die ich oben schon erwähnt habe. Die sind neben Props eines der wichtigsten Konzepte in React. Immer wenn sich was am State ändert, wird die Komponent neu gerendert, um die Änderungen anzuzeigen. Das nutzen wir hier aus, um einen Request abzuschicken, und den State erst dann zu setzen, wenn eine Antwort (erfolgreich oder nicht) vom Server kommt. Dadurch wird erneut gerendert, was dann dazu führt, dass sich die Ergebnisse automatisch updaten.